pkg/handlers/valkey/valkey.go

```diff
@@ -118,11 +118,10 @@ func (h ValkeyHandler) Apply(ctx context.Context, application *aiven_nais_io_v1.
 			fmt.Sprintf("%s_%s", redis.RedisHost, envVarSuffix):     addresses.Valkey.Host,
 			fmt.Sprintf("%s_%s", redis.RedisURI, envVarSuffix):      strings.Replace(addresses.Valkey.URI, "valkeys", "rediss", 1),
 		})
+		controllerutil.AddFinalizer(valkeySecret, constants.AivenatorFinalizer)
```

Comment:
Kunne vi lagt `AddFinalizer` inn i `initSecret`? For `initSecret` er jo lik for alle sammen, så der burde vi klare å få en felles funksjon.